packages/dev/core/src/Rendering/fluidRenderer/fluidRenderer.ts

```diff
@@ -0,0 +1,525 @@
+                }
+            } else if (!ps.renderAsFluid) {
+                this._renderObjects[index].object.dispose();
+                this._renderObjects.splice(index, 1);
+            }
+        }
+        this._removeUnusedTargetRenderers();
+        this._initialize();
+    }
+
+    private _removeUnusedTargetRenderers(): boolean {
+        const indexes: { [id: number]: boolean } = {};
+
+        for (let i = 0; i < this._renderObjects.length; ++i) {
+            const targetRenderer = this._renderObjects[i].targetRenderer;
+            indexes[this._targetRenderers.indexOf(targetRenderer)] = true;
+        }
+
+        let removed = false;
+        const newList: Array<FluidRenderingTargetRenderer> = [];
+        for (let i = 0; i < this._targetRenderers.length; ++i) {
+            if (!indexes[i]) {
+                this._targetRenderers[i].dispose();
+                removed = true;
+            } else {
+                newList.push(this._targetRenderers[i]);
+            }
+        }
+
+        if (removed) {
+            this._targetRenderers.length = 0;
+            this._targetRenderers.push(...newList);
+        }
+
+        return removed;
+    }
+
+    private static _IsParticleSystemObject(obj: FluidRenderingObject): obj is FluidRenderingObjectParticleSystem {
+        return !!(obj as FluidRenderingObjectParticleSystem).particleSystem;
+    }
+
+    private static _IsVertexBufferObject(obj: FluidRenderingObject): obj is FluidRenderingObjectVertexBuffer {
+        return (obj as FluidRenderingObjectVertexBuffer).getClassName() === "FluidRenderingObjectVertexBuffer";
+    }
+
+    private _getParticleSystemIndex(ps: IParticleSystem): number {
+        for (let i = 0; i < this._renderObjects.length; ++i) {
+            const obj = this._renderObjects[i].object;
+            if (FluidRenderer._IsParticleSystemObject(obj) && obj.particleSystem === ps) {
+                return i;
+            }
+        }
+
+        return -1;
+    }
+
+    private _getVertexBufferIndex(vb: VertexBuffer): number {
+        for (let i = 0; i < this._renderObjects.length; ++i) {
+            const obj = this._renderObjects[i].object;
+            if (FluidRenderer._IsVertexBufferObject(obj) && obj.vertexBuffers[VertexBuffer.PositionKind] === vb) {
+                return i;
+            }
+        }
+
+        return -1;
+    }
+
+    private _initialize(): void {
+        for (let i = 0; i < this._targetRenderers.length; ++i) {
+            this._targetRenderers[i].dispose();
+        }
+
+        const cameras: Map<Camera, [Array<FluidRenderingTargetRenderer>, { [key: string]: FluidRenderingCopyDepthTexture }]> = new Map();
+
+        for (let i = 0; i < this._targetRenderers.length; ++i) {
+            const targetRenderer = this._targetRenderers[i];
+
+            targetRenderer._initialize();
+
+            if (targetRenderer.camera && targetRenderer._renderPostProcess) {
+                let list = cameras.get(targetRenderer.camera);

```

Comment:
It will return `undefined` the first time we encounter the camera, but when it's `undefined` we are doing what's inside the `if`:
```typescript
if (!list) {
    list = [[], {}];
    cameras.set(targetRenderer.camera, list);
}
list[0].push(targetRenderer);
```
So the 2nd time we encounter the same camera (if/when multiple target renderers are using the camera) we will get this list and will add the `targetRenderer` to this list.